Userland/Applications/MailSettings/MailSettingsWindow.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2021, The SerenityOS developers
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+
+#include "MailSettingsWindow.h"
+#include <Applications/MailSettings/MailSettingsWindowGML.h>
+#include <LibGUI/TextEditor.h>
+#include <LibCore/ConfigFile.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/CheckBox.h>
+#include <LibGUI/SpinBox.h>
+#include <LibGUI/BoxLayout.h>
+#include <LibGUI/Widget.h>
+#include <LibGUI/TabWidget.h>
+#include <unistd.h>
+
+void MailSettingsWindow::reset_default_values()
+{
+    m_server_inputbox->set_text(server);
+    m_port_spinbox->set_value(port);
+    m_tls_checkbox->set_checked(tls);
+    m_email_inputbox->set_text(email);
+    update();
+}
+
+void MailSettingsWindow::write_values()
+{
+    server = m_server_inputbox->get_text();
+    port = m_port_spinbox->value();
+    tls = m_tls_checkbox->is_checked();
+    email = m_email_inputbox->get_text();
+
+    config->write_entry("Connection", "Server", server);
+    config->write_num_entry("Connection", "Port", port);
+    config->write_bool_entry("Connection", "TLS", tls);
+    config->write_entry("Connection", "Username", email);
+    config->sync();
+}
+
+MailSettingsWindow::MailSettingsWindow()
+{
+    config = Core::ConfigFile::get_for_app("Mail");
+    if (unveil(config->filename().characters(), "rwc") < 0) {
+        perror("unveil");
+        GUI::Application::the()->quit();
+    }
+
+    auto& main_widget = set_main_widget<GUI::Widget>();
+    main_widget.set_fill_with_background_color(true);
+    main_widget.set_layout<GUI::VerticalBoxLayout>();
+    main_widget.layout()->set_margins({ 4, 4, 4, 4 });
+    main_widget.layout()->set_spacing(6);
+
+    auto& tab_widget = main_widget.add<GUI::TabWidget>();
+    auto& mail_widget = tab_widget.add_tab<GUI::Widget>("Mail");
+    mail_widget.load_from_gml(mail_settings_window_gml);
```

Comment:
The changes in this commit are not usable on their own because the GML file is empty. Can you please squash the commits so that:

1. Each of them is buildable and usable.
2. Commits which fix problems which were introduced in an earlier commit in the PR are squashed with that commit.

Reply:
I didn't realize I could do that, I'll figure it out and do it